gitmanager.py

```diff
@@ -306,20 +307,56 @@ def remove_from_blacklist(cls, item, username, blacklist_type="", code_privilege
     @classmethod
     def merge_pull_request(cls, pr_id, comment=""):
         response = requests.get("https://api.github.com/repos/{}/pulls/{}".format(GlobalVars.bot_repo_slug, pr_id))
-        if not response:
+        files = requests.get("https://api.github.com/repos/{}/pulls/{}/files".format(GlobalVars.bot_repo_slug, pr_id))
+        if not response or not files:
             raise ConnectionError("Cannot connect to GitHub API")
         pr_info = response.json()
-        if pr_info["user"]["login"] != "SmokeDetector":
+        files_info = files.json()
+        if pr_info["user"]["login"] != "Daniil-SD":
             raise ValueError("PR #{} is not created by me, so I can't approve it.".format(pr_id))
         if "<!-- METASMOKE-BLACKLIST" not in pr_info["body"]:
             raise ValueError("PR description is malformed. Blame a developer.")
         if pr_info["state"] != "open":
             raise ValueError("PR #{} is not currently open, so I won't merge it.".format(pr_id))
+        string = pr_info["title"]
+        file = files_info[0]["filename"]
+        print(file)
         ref = pr_info['head']['ref']
+        string1 = regex.match(r".*?: \S+ (.*?)(?:\Z|\s*\(\?#)", string).group(1)
+        username = regex.match(r".*(?=:)", string).group(0)
+        if file == "blacklisted_websites.txt":
+            blacklist_type = Blacklist.WEBSITES
+        elif file == "bad_keywords.txt":
+            blacklist_type = Blacklist.WEBSITES
+        elif file == "blacklisted_usernames.txt":
+            blacklist_type = Blacklist.USERNAMES
+        elif file == "blacklisted_numbers.txt":
+            blacklist_type = Blacklist.NUMBERS
+        elif file == "watched_keywords.txt":
+            blacklist_type = Blacklist.WATCHED_KEYWORDS
+        elif file == "watched_numbers.txt":
+            blacklist_type = Blacklist.WATCHED_NUMBERS
+        else:
+            raise CmdException('GitManager: blacklist is not recognized. Blame a developer.')
+        try:
+            blacklister = Blacklist(blacklist_type)
+            blacklist_file_name = blacklist_type[0]
+            item_to_blacklist = string
+            if blacklist_type in {Blacklist.WATCHED_KEYWORDS, Blacklist.WATCHED_NUMBERS}:
+                op = 'watch'
+                item = item_to_blacklist
+                item_to_blacklist = "\t".join([now, username, item])
+            else:
+                op = 'blacklist'
+                item = item_to_blacklist
 
+            exists, line = blacklister.exists(item_to_blacklist)
+            if exists:
+                raise RuntimeError('Already {}ed on line {} of {}'.format(op, line, blacklist_file_name))
+        except Exception:
+            print(sys.format_exc())
```

Comment:
While it's OK to have the above code here for testing, this is effectively duplicating what's in `add_to_blacklist()` (which is what is needed), but it would be better to pull most of this out of both functions into a helper function, so it's only performed in one location, rather than having to maintain nearly identical code in two places.